### ISSUE TITLE
Civicrm crashed during upgrade

### ISSUE THREAD

**Issue by partibrejker2** [mlutfy/ca.bidon.reporterror#30]
Hi, today i tried to upgrade extension and it crashed my Civicrm.
I can access Joomla normally, but when i click on Civicrm, it shows HTTP ERROR 500.

Civicrm is 4.7
Joomla 3.8.7.
PHP Version 7.0.30

**Comment by partibrejker2** [comment]
update - cannot access administrator part of Joomla. After login just blank screen.

**Comment by mlutfy** [comment]
hi @partibrejker2, did you update to the packaged version with the "vendor" directory? c.f. https://github.com/mlutfy/ca.bidon.reporterror/releases/tag/3.1 (reporterror-3.1.tar.gz)

If not, what is the method you currently use? I guess it might be good to better safeguard against this.

**Comment by kcristiano** [comment]
@mlutfy I updated via the CiviCRM extension screen the first time, then I manually retrieved the 3.3 package from the releases page.   

I will try this again on the staging site we have.    I did this update live on production.

**Comment by partibrejker2** [comment]
Hi Mlufty

like cristiano i updated via CiviCRM extension screen. And the error is the same.
I would appreciate if somebody could help me at least revert this.


`[09-May-2018 10:29:29 Europe/*] PHP Warning:  require_once(*/media/civicrm/ext/ca.bidon.reporterror/vendor/autoload.php): failed to open stream: No such file or directory in */media/civicrm/ext/ca.bidon.reporterror/reporterror.php on line 9
[09-May-2018 10:29:29 Europe/*] PHP Fatal error:  require_once(): Failed opening required '*/media/civicrm/ext/ca.bidon.reporterror/vendor/autoload.php' (include_path='.:/*/administrator/components/com_civicrm/civicrm:/*/administrator/components/com_civicrm/civicrm/packages:/*/administrator/components/com_civicrm/civicrm/vendor/tecnickcom:/*/administrator/components/com_civicrm/civicrm/vendor/phpseclib/phpseclib/phpseclib:/*/administrator/components/com_civicrm/civicrm/vendor/pear/pear_exception:/*/administrator/components/com_civicrm/civicrm/vendor/pear/auth_sasl:/*/administrator/components/com_civicrm/civicrm/vendor/pear/net_socket:/*/administrator/components/com_civicrm/civicrm/vendor/pear/net_smtp:/home/evoli in /*/media/civicrm/ext/ca.bidon.reporterror/reporterror.php on line 9`

**Comment by kcristiano** [comment]
I just tried a direct download and I don't see a vendor directory in the extension.

**Comment by mlutfy** [comment]
> I just tried a direct download and I don't see a vendor directory in the extension.

Using: https://github.com/mlutfy/ca.bidon.reporterror/releases/download/3.1/reporterror-3.1.tar.gz ?

**Comment by kcristiano** [comment]
I should have included the link.

I used https://github.com/mlutfy/ca.bidon.reporterror/archive/3.1.zip  

not https://github.com/mlutfy/ca.bidon.reporterror/releases/download/3.1/reporterror-3.1.tar.gz

Off to try again.    Still unsure why the CiviCRM updater failed.

**Comment by partibrejker2** [comment]
so, what do i do to get my website back? :)

**Comment by mlutfy** [comment]
@partibrejker2 Update manually the extension using: https://github.com/mlutfy/ca.bidon.reporterror/releases/download/3.1/reporterror-3.1.tar.gz

**Comment by partibrejker2** [comment]
i have never done it before.
Do i just copy the files?
If so, do i just copy them to : media/civicrm/ext ?

Do i have to do anything in the database?

thank you for your help

**Comment by mlutfy** [comment]
yep, that should work.

Worst case, you can also try deleting the `ca.bidon.reporterror` directory, then go to Admin > System > Extension, click 'refresh', then re-install (I updated the link there).

**Comment by partibrejker2** [comment]
Thx,

i had to delete it.
Now:

![image](https://user-images.githubusercontent.com/38113193/39820247-a9a5094c-53a5-11e8-865e-93c62311b3c2.png)

**Comment by mlutfy** [comment]
Getting there .. ;)

Can you refresh your extension list and try again? I updated for a zip file.

**Comment by mattwire** [comment]
@mlutfy Running `composer install` works when doing a git clone as the vendor dir is not included in the github repository.  This caught me out as it's a new dependency.  Could you add something to the readme?

**Comment by mlutfy** [comment]
@mattwire good idea, added a2e4081f3ce9fe6f09f9ef8504780d278110b506 (although I suspect most people don't read that. It would be better to check before requiring the autoload, and if it's missing show a status message?)

**Comment by mattwire** [comment]
> It would be better to check before requiring the autoload, and if it's missing show a status message

@mlutfy Yes, good idea. I think some other extensions have handled things in a similar way (eg. mosaico).
